esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -136,15 +140,25 @@ def _roll_cube_data(cube, shift, axis):
     return cube
 
 
-def _save_variable(cube, var, outdir, year, proj):
+def _save_variable(cube, var, outdir, proj, zip='False'):
     """Saver function."""
     # CMOR standard
-    time_suffix = '-'.join([str(year) + '01', str(year) + '12'])
+    yrbeg=cube.coord('time').cell(0).point.strftime('%Y')
+    mobeg=cube.coord('time').cell(0).point.strftime('%m')
+    yrend=cube.coord('time').cell(-1).point.strftime('%Y')
+    moend=cube.coord('time').cell(-1).point.strftime('%m')
+    time_suffix = '-'.join([yrbeg + mobeg, yrend + moend])
+
```

Comment:
this is convoluted to say the least; the previous implementation allows for arbitrary years as passed by the cmorizer script; please revert

Reply:
Beside the coding style, I don't agree. 
The previous implementation was using a prescribed field passed by the cmorizer, which is first used to adjust the time in `_convert_timeunits`, specifically fix a bug of WOA dataset (note that the else case was also missing!). 
So after this step the cube has the correct time and there is no need to drive around the year value in the `_save_variable` function only to define the output filename, while it seems more straightforward to define the date within the `_save_variable` function.

Reply:
the coding style request applies all through your PR, please see the Codacy results at the bottom, if you inspect the problems reported you will find a list of style complaints from PEP analyzer. The problem here lies within the fact that the cmorization can be done on multiple cubes that are individually cmorized from the cmorizer script but saved as a concatenated listso hence the need to have a first and a last year - this current implementation is dodgy indeed but I have changed it in a new PR, so rest assured, it will improve soon :grin:

Reply:
ok! 
I'll revert the changes of the save function in `utilities.py` script and keep them only within the cmorizer script.